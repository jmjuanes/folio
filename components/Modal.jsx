import React from "react";
import classNames from "classnames";
import {CloseIcon} from "@josemi-icons/react";

export const Modal = props => (
    <React.Fragment>
        <div className="fixed w-full h-full bg-white o-70 z-9 top-0 left-0" />
        <div data-testid="modal" className="fixed w-full h-full flex items-center justify-center z-10 top-0 left-0">
            <div className={classNames(props.className, "rounded-lg border border-neutral-200 bg-white w-full p-6 shadow-sm")}>
                <div className="flex items-center justify-between mb-3 select-none">
                    <div data-testid="modal-title" className="flex items-center text-lg font-bold text-neutral-900">
                        <span>{props.title}</span>
                    </div>
                    <div data-testid="modal-close" className="flex group" onClick={props.onClose}>
                        <div className="flex text-2xl text-neutral-500 group-hover:text-neutral-900 cursor-pointer">
                            <CloseIcon />
                        </div>
                    </div>
                </div>
                <div data-testid="modal-content">
                    {props.children}
                </div>
            </div>          
        </div>
    </React.Fragment>
);

Modal.defaultProps = {
    className: "max-w-lg",
    title: "",
    onClose: null,
};
