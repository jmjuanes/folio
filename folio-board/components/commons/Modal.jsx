import React from "react";
import {CloseIcon} from "@mochicons/react";

export const Modal = props => (
    <React.Fragment>
        <div className="position:fixed w:full h:full bg:dark-700 o:50 z:9 top:0 left:0" />
        <div className="position:fixed w:full h:full d:flex items:center justify:center z:10 top:0 left:0">
            <div className="r:xl bg:light-100 w:full p:8" style={{maxWidth: props.maxWidth}}>
                {!!props.onClose && (
                    <div className="d:flex items:center justify:end">
                        <div className="d:flex r:full bg:light-300 bg:light-500:hover p:2" onClick={props.onClose}>
                            <div className="d:flex text:2xl text:dark-100 cursor:pointer">
                                <CloseIcon />
                            </div>
                        </div>
                    </div>
                )}
                {props.children}
            </div>          
        </div>
    </React.Fragment>
);

Modal.defaultProps = {
    maxWidth: "500px",
    onClose: null,
};
