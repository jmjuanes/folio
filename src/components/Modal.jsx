import React from "react";
import {CloseIcon} from "@mochicons/react";

export const Modal = props => (
    <React.Fragment>
        <div className="fixed w-full h-full bg-gray-900 o-50 z-9 top-0 left-0" />
        <div className="fixed w-full h-full flex items-center justify-center z-10 top-0 left-0">
            <div className="rounded-xl border border-gray-300 bg-white w-full p-8" style={{maxWidth: props.maxWidth}}>
                {!!props.onClose && (
                    <div className="flex items-center justify-end">
                        <div className="flex r-full bg-gray-100 hover:bg-gray-200 p-2" onClick={props.onClose}>
                            <div className="flex text-4xl text-gray-500 cursor-pointer">
                                <CloseIcon />
                            </div>
                        </div>
                    </div>
                )}
                {props.children}
            </div>          
        </div>
    </React.Fragment>
);

Modal.defaultProps = {
    maxWidth: "500px",
    onClose: null,
};
