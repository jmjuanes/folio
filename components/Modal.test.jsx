import {fireEvent, render, screen, act, waitFor} from "@testing-library/react";
import {Modal} from "./Modal.jsx";

jest.mock("@josemi-icons/react", () => ({
    CloseIcon: jest.fn(() => "CLOSE_ICON"),
}));

describe("Modal", () => {
    it("should render", async () => {
        render(<Modal />);
        await waitFor(() => {
            return expect(screen.getByTestId("modal")).toBeDefined();
        });
    });

    it("should render the title", async () => {
        const title = "MODAL_TITLE";
        render(<Modal title={title} />);
        await waitFor(() => {
            return expect(screen.getByTestId("modal-title").textContent).toEqual(title);
        });
    });

    it("should execute the provided function when clicking the close button", async () => {
        const onClose = jest.fn();
        render(<Modal onClose={onClose} />);
        await waitFor(() => {
            return expect(screen.getByTestId("modal")).toBeDefined();
        });
        act(() => {
            fireEvent.click(screen.getByTestId("modal-close"));
        });
        expect(onClose).toHaveBeenCalled();
    });

    it("should render the provided content", async () => {
        const content = "MODAL_CONTENT";
        render(<Modal>{content}</Modal>);
        await waitFor(() => {
            return expect(screen.getByTestId("modal")).toBeDefined();
        });
        expect(screen.getByTestId("modal-content").textContent).toEqual(content);
    });
});
