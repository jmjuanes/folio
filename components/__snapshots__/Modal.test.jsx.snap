// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render 1`] = `
Array [
  <div
    className="fixed w-full h-full bg-gray-900 o-50 z-9 top-0 left-0"
  />,
  <div
    className="fixed w-full h-full flex items-center justify-center z-10 top-0 left-0"
  >
    <div
      className="rounded-xl border border-gray-300 bg-white w-full p-8"
      style={
        Object {
          "maxWidth": "500px",
        }
      }
    >
      Content
    </div>
  </div>,
]
`;
